RecoEgamma/ElectronIdentification/interface/ElectronMVAEstimatorRun2Spring16GeneralPurpose.h

```diff
@@ -0,0 +1,154 @@
+    float mcEventWeight; // 29
+    float mcCBmatchingCategory; // 30
+    
+  };
+  
+  // Constructor and destructor
+  ElectronMVAEstimatorRun2Spring16GeneralPurpose(const edm::ParameterSet& conf);
+  ElectronMVAEstimatorRun2Spring16GeneralPurpose(); // For Reflex. Not to be used in reality
+  ElectronMVAEstimatorRun2Spring16GeneralPurpose(const std::string &mvaTag, const std::string &conversionsTag = "reducedEgamma:reducedConversions", const std::string &beamspotTag = "offlineBeamSpot");
+  void init(const std::vector <std::string> weightFileNames);
+  ~ElectronMVAEstimatorRun2Spring16GeneralPurpose();
+
+
+
+  // Calculation of the MVA value (VID accessor)
+  float mvaValue( const edm::Ptr<reco::Candidate>& particle, const edm::Event&) const override;
+  // Calculation of the MVA value (fwlite-compatible accessor)
+  float mvaValue( const reco::GsfElectron * particle, const edm::EventBase &) const ;
+  // Calculation of the MVA value (bare version)
+  float mvaValue( const int iCategory, const std::vector<float> & vars) const ;
+
+  // Utility functions
+  std::unique_ptr<const GBRForest> createSingleReader(const int iCategory, 
+                                                      const edm::FileInPath &weightFile);
+
+  virtual int getNCategories() const override { return nCategories; }
+  bool isEndcapCategory( int category ) const;
+  virtual const std::string& getName() const override final { return _name; }
+  virtual const std::string& getTag() const override final { return _tag; }
+
+  // Functions that should work on both pat and reco electrons
+  // (use the fact that pat::Electron inherits from reco::GsfElectron)
+  std::vector<float> fillMVAVariables(const edm::Ptr<reco::Candidate>& particle, const edm::Event&) const override;
+  std::vector<float> fillMVAVariables( const reco::GsfElectron * particle, const edm::Handle<reco::ConversionCollection> conversions, const reco::BeamSpot *beamSpot) const ;
+  int findCategory( const edm::Ptr<reco::Candidate>& particle) const override;
+  int findCategory( const reco::GsfElectron * particle) const ;
+  // The function below ensures that the variables passed to MVA are 
+  // within reasonable bounds
+  void constrainMVAVariables(AllVariables&) const;
+
+  // Call this function once after the constructor to declare
+  // the needed event content pieces to the framework
+  void setConsumes(edm::ConsumesCollector&&) const override final;
+  // Call this function once per event to retrieve all needed
+  // event content pices
+  
+ private:
+
+  // MVA name. This is a unique name for this MVA implementation.
+  // It will be used as part of ValueMap names.
+  // For simplicity, keep it set to the class name.
+  const std::string _name = "ElectronMVAEstimatorRun2Spring16GeneralPurpose";
+  // MVA tag. This is an additional string variable to distinguish
+  // instances of the estimator of this class configured with different
+  // weight files.
+  const std::string _tag;
```

Comment:
There are still prohibited variables names that start with "_" here and on lines 136-149.